homeassistant/components/justnimbus/__init__.py

```diff
@@ -3,14 +3,18 @@
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
 
 from .const import DOMAIN, PLATFORMS
 from .coordinator import JustNimbusCoordinator
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up JustNimbus from a config entry."""
-    coordinator = JustNimbusCoordinator(hass=hass, entry=entry)
+    try:
+        coordinator = JustNimbusCoordinator(hass=hass, entry=entry)
+    except KeyError as err:
+        raise ConfigEntryAuthFailed(err) from err
```

Comment:
Test this case by adding a mock config entry with old data and set it up. Then check that there's a reauth flow in progress. Add this test to `test_init.py`.